common/src/lib.rs

```diff
@@ -326,6 +360,41 @@ pub fn remove_waiting_message(prog_id: H256, msg_id: H256) -> Option<(Message, u
     msg
 }
 
+pub struct WaitingMessageIterator(PrefixIterator<(Message, u32)>);
+
+fn decode_message_tuple(_: &[u8], value: &[u8]) -> Result<(Message, u32), codec::Error> {
+    <(Message, u32)>::decode(&mut &*value)
+}
+
+impl WaitingMessageIterator {
+    pub fn new(destination_program_id: H256, previous_message_id: Option<H256>) -> Self {
+        let key_prefix = match previous_message_id {
+            None => wait_prefix(destination_program_id),
+            Some(id) => wait_key(destination_program_id, id),
```

Comment:
What is the purpose of this iterator? Is it supposed to iterate trough all waitlisted messages for a given program (potentially, starting from some specific message id)?
Based on this implementation, it seems that if you provide a `previous_message_id` the prefix will be set to the encoded combination of `WAIT_PREFIX::program_id::message_id` and there is at most one such key in the storage. Is this intentional?
Maybe the `prefix` should only contain the `program_id`, regardless of the starting point (which is determined by the `previous_message_id`)?

Reply:
Yeah, this is an error. Good catch!